lib/services/ios-log-filter.ts

```diff
@@ -1,65 +1,86 @@
 	private partialLine: string = null;
 
-	constructor($loggingLevels: Mobile.ILoggingLevels,
+	constructor(
+		private $loggingLevels: Mobile.ILoggingLevels,
 		private $fs: IFileSystem,
 		private $projectData: IProjectData) {
-		super($loggingLevels);
+		this.projectName = this.$projectData.projectName;
 	}
 
 	public filterData(data: string, logLevel: string, pid?: string): string {
-		data = super.filterData(data, logLevel, pid);
-		if (pid && data && data.indexOf(`[${pid}]`) === -1) {
-			return null;
+		const specifiedLogLevel = (logLevel || '').toUpperCase();
+
+		if (specifiedLogLevel !== this.$loggingLevels.info || !data) {
+			return data;
 		}
 
-		if (data) {
-			const skipLastLine = data[data.length - 1] !== "\n";
-			const lines = data.split("\n");
-			let result = "";
-			for (let i = 0; i < lines.length; i++) {
-				let line = lines[i];
-				if (i === 0 && this.partialLine) {
-					line = this.partialLine + line;
-					this.partialLine = null;
-				}
-				if (line.length < 1 ||
-					line.indexOf("SecTaskCopyDebugDescription") !== -1 ||
-					line.indexOf("NativeScript loaded bundle") !== -1 ||
-					(line.indexOf("assertion failed:") !== -1 && data.indexOf("libxpc.dylib") !== -1)) {
-					continue;
-				}
-				// CONSOLE LOG messages comme in the following form:
-				// <date> <domain> <app>[pid] CONSOLE LOG file:///location:row:column: <actual message goes here>
-				// This code removes unnecessary information from log messages. The output looks like:
-				// CONSOLE LOG file:///location:row:column: <actual message goes here>
-				if (pid) {
-					if (line.indexOf(`[${pid}]: `) !== -1) {
-						const pidRegex = new RegExp(`^.*\\[${pid}\\]:\\s(?:\\(NativeScript\\)\\s)?`);
-						line = line.replace(pidRegex, "").trim();
-						this.getOriginalFileLocation(line);
-						result += this.getOriginalFileLocation(line) + "\n";
-					}
+		const chunkLines = data.split('\n');
+		const skipLastLine = chunkLines.length > 1 ? data[data.length - 1] !== "\n" : false;
+		let output = "";
+		for (let i = 0; i < chunkLines.length; i++) {
+			let currentLine = chunkLines[i];
 
-					continue;
-				}
-				if (skipLastLine && i === lines.length - 1 && lines.length > 1) {
-					this.partialLine = line;
-				} else {
-					result += this.getOriginalFileLocation(line) + "\n";
-				}
+			// Legacy filter moved to preFilter
+			if (this.preFilter(data, currentLine)) {
+				continue;
+			}
+
+			const matchResult = this.appOutputRegex.exec(currentLine);
+
+			if (matchResult && matchResult.length > 1) {
+				// Check if the name of the app equals the name of the CLI project and turn on the filter if not.
+				this.filterActive = matchResult[1] !== this.projectName;
 			}
-			return result;
+
+			if (this.filterActive) {
+				continue;
+			}
+
+			if (this.partialLine) {
+				currentLine = this.partialLine + currentLine;
+				this.partialLine = undefined;
+			}
+
+			if (i === chunkLines.length - 1 && skipLastLine) {
+				this.partialLine = currentLine;
```

Comment:
Would be grate if we could persist this per device identifier so that we don't mix lines from different devices. (This is not mandatory as it was the same way before).

Reply:
As far as I know, there is no way of receiving log messages from parallel processes with the latest version of CLI.